### DISCUSSION TITLE
Scoring the spectrum of deposited energy in a thin silicon sensor, radiative corrections and high energy simulations

### DISCUSSION THREAD

**Discussion by Jaap1957** [nrc-cnrc/EGSnrc#826] [Physics] [answered]
Hi, 

I want to simulate energy deposition in a thin (15mu) SI slab from a 0.3MeV electron beam.
I found already many answers in the EGS manual and in egsinp files like 
 EGS_HOME\sprrznrc\sprrznrc_template.egsinp 

A few remaining questions:

1  default setting for Electron Impact Ionization is Off. 
 would it be a good idea to put one of the "on" options here?

Further, when comparing the output of  egs_phd for 
ge-example and icrm-hpge-example

running ge-example i get
***************** Warning: 
 You are trying to use radiative Compton corrections
 without having included rad_compton1.mortran
Turning radiative Compton corrections OFF ...

2 Should i include radiative Compton corrections for a 0.3MeV electron beam ?
2A the makefile for the pp applications is quite different from the one where the hint points to.
How to include the rad_compton1.mortran functionality in a egs pp application?

3  Bound Compton scattering norej, simple or On: Which one to choose?

Thanks !

**Comment by rtownson** [comment]
Yes, it's a good idea to turn EII on. I think that radiative Compton corrections will have a small impact at 0.3 MeV but it's good practice to check.

See here for some instructions:
https://www.reddit.com/r/EGSnrc/comments/ecuql7/radiative_compton_corrections_on/

Bound Compton scattering `On` is a good choice.

**Reply by ftessier** [reply]
Reminder: radiative Compton corrections requires compilations of extra source files, i.e., just turning "On" the option is not sufficient. See the [EGSnrc manual](https://nrc-cnrc.github.io/EGSnrc/doc/pirs701-egsnrc.pdf) page 46.

**Reply by ftessier** [reply]
Oh, I see Reid has already pointed to instructions on reddit! 👍🏻

**Comment by Jaap1957** [comment]
Thanks Mr Townson, that solves the program side of the problem.
I simulated with 1 layer of 32 mu Si , 2 MeV e- pencil beam, 1e6 histories.
That sitaution is one of the entries in the table at pg 23 of
Straggling in thin silicon detectors, Hans Bichsel, Reviews of Modern Physics, Vo!. 60, No. 3, July 1988
He gives : 7.1 keV deposited, 5.5 keV width of pulse height.

EGSnrc produces absorbed dose per e- = 9.9 keV
The pulse width spectrum , obtained with egs_phd gives
avg = 9.4 std = 4.4
Here the pulse height spectrum i get:

![afbeelding](https://user-images.githubusercontent.com/96918927/150829127-7d3c416d-165c-41fe-af58-df0e655de3eb.png)

1 Why does the pulse height spectrum cut off at 8.5 keV ?

2 Why does EGSnrc give 20% more deposited energy than literature ?

**Reply by rtownson** [reply]
1. What's your AE and ECUT? You'll want both of these at 0.512 MeV, the minimum.
2. That's not an easy question. You would need to consider the differences in the MC codes used, the cross sections and other data, the physics, etc.

**Reply by Jaap1957** [reply]
You were right, the Ecut was at 0.521, Pcut at 0.01
Here again the previous graph, now in semi logy:
![afbeelding](https://user-images.githubusercontent.com/96918927/150867805-5158abdf-0d42-4a8d-9d6d-8e771d0b3be0.png)

When I change to Ecut = ae =  0.512 and Pcut = ap = 0.005:
![afbeelding](https://user-images.githubusercontent.com/96918927/150868630-86ebf2ca-1b81-4bcf-9fa1-ab7b3aea0930.png)

The average dose of 9.5 still there. The graph indeed shifts somewhat to the left.
According to litreature i should expect about 7.

I will go on and try different settings. Do you have a clue why lieterature would quote 20% less dose than EGS?

**Reply by ftessier** [reply]
Isn't it interesting to review theory and older data with modern methods? The article itself points to the some calibration, scaling and adjustments required to match theory and experiments in this 1988 paper. In particular, I note:

> An adjustment of the energy scales of the experimental data was made (i.e., the energy-loss scale was shifted to give the value of <sub>x</sub>Δ<sub>p</sub> equal to <sub>i</sub>Δ<sub>p</sub>).

I would suggest first looking at ratios of the peak energy obtained with different thicknesses or energies. If ratios are correct, then one may try to determine biases in the experimental results (or the simulation). Typically, it is difficult to match simulation data with expreimental measurement, because of unknowns on the experimental front (the simulation is always an _ideal_ scenario). Using ratios is more robust because some of the unknowns normalize out. One very common example is the incident electron beam current, for example, in an x-ray tube simulation.

Let us know what you find!

**Reply by Jaap1957** [reply]
Thank you for your comment Mr Tessier !

Classic mistake: Bichsel quotes "betagamma": I translated 3914 into 2MeV but that should have been 2Gev .... Since 2MeV is already somewhat relativistic the difference is not large.

Still my question is: Why is there a cutoff at low energy. 
The cutoff shifts with material thickness. Here the curve for 174mu Si thickness:
cutoff around 38keV.
![afbeelding](https://user-images.githubusercontent.com/96918927/150915118-08214a64-0dd7-438d-85d8-abf78792a7f7.png)

**Reply by Jaap1957** [reply]
setting thickness to 0.000015 , at 0.3MeV, the graph is
![afbeelding](https://user-images.githubusercontent.com/96918927/150929153-2d7f8ae2-5e2f-4ec4-ab30-88b1124eee33.png)

There are edges at ~ 1 and 1.8 keV
At 0.0015 cm i get
![afbeelding](https://user-images.githubusercontent.com/96918927/150929429-851c1329-1559-4345-a413-4eb1b0536b4e.png)

The 'edges'  now are at 5.5, 6.2 keV.

What are these edges? Clearly not a material 'K edge' or so, otherwise they would be stable with thickness.

**Reply by ftessier** [reply]
Recall that the transport cutoffs is 1 keV, so there will be a jump at 1 keV since many particles reaching this energy will stop abruptly and deposit their energy on the spot (which is an artefact of the cutoff approximation). I don't know why the edges are moving with thickness. Are you sure you are scoring dose in the silicon, and not in the transmitted region?

Also, at these thicknesses you may want to run the simulation in single-scattering mode, which is more efficient when the number of electron mean free paths to the closest boundary is ~ 3 (see the [EGSnrc manual](https://nrc-cnrc.github.io/EGSnrc/doc/pirs701-egsnrc.pdf), p. 101). To calculate in single-scattering mode, set the transport parameter `Skin depth for BCA = 1e10` or some other very large value.

**Reply by Jaap1957** [reply]
Yes, 1 keV. Why do i see a cutoff at 0.06 keV, much lower, or at 4.4 keV, much higher?

Attached my input file, for  0.000015  cm, 0.3 MeV.
After doing the tutor7pp example I never feel confortable anymore about region numbers ...
[slabT6.egsinp.txt](https://github.com/nrc-cnrc/EGSnrc/files/7933872/slabT6.egsinp.txt)
I use the egs_phd code, adapted with your help to include radidative compton correciton.

Here output of egs_view 
![afbeelding](https://user-images.githubusercontent.com/96918927/150986015-17e7f811-98b4-42df-ab87-8eca097483d0.png)

**Reply by ftessier** [reply]
Thank you for the input file. I am able to reproduce your data. The 1 keV edge is the `ae` threshold artifact (secondary electrons are never created below `ae`, hence the depletion), and as expected it moves in stride with the value of `ae`:

![montage](https://user-images.githubusercontent.com/1444241/151023017-dfe0fb17-ac5f-47c1-b80a-2fe8e8fcf2e1.png)

**The first peak is located at the minimum energy it takes for the electron to cross the slab** (with only sub-threshold events and no deflections, i.e., straight CSDA); this is why it increases with thickness (or more exactly mass thickness as seen in the tungsten data on the right). Keep in mind that what is scored here is the total energy deposition for the entire shower (one incident electron), not the energy of every single energy deposition event.

The edge at 2 keV in silicon, I verified quickly that it depends on Z, so it appears "real" (although perhaps it _could_ be an artefact in the data tabulations?):

![slab-03-material-dependence](https://user-images.githubusercontent.com/1444241/151023688-69ed48fd-beea-401f-806e-44581efe5ce1.png)

**Reply by ftessier** [reply]
Out of curiosity, here _is_ the spectrum of the _individual_ energy deposition events, for silicon slabs of 150 nm and 1500 nm. As you can see, the energy deposition physics remains largely the same, however the first peak due to CSDA electrons going straight through is displaced to the right, and is much less pronounced, as expected (electrons are less likely to go straight trough for a thicker sample). The `ae` artifact peak is clearly visible at 0.001 MeV here.

![slab-04-all-edep-events](https://user-images.githubusercontent.com/1444241/151032329-acba3ddc-05d3-4bf5-ba34-97d790e5207f.png)

**Reply by Jaap1957** [reply]
To finish this exercise, here's the table of Hans Bichsel, where i added my first steps on MC ground.
see the attached excel file.
[Bichsel.xlsx](https://github.com/nrc-cnrc/EGSnrc/files/7940518/Bichsel.xlsx)

The maximum correlates the best with the value of the literature.

Indeed this is old literature already ... 
What kind of new experimental or MC data would you suggest to look at in the field of :
- 100 - 300 keV
- Si layers 15 - 100 mu
- Conversion to e-hole pairs in EPI
?

**Reply by ftessier** [reply]
Thank you for sharing your results. Question: did you extend the density effect corrections above 1 GeV?

Fair agreement (is it within uncertainties?). As @mainegra pointed out, we have not looked at high-energy simulations with EGSnrc for a long while... But typically in that range all approximations improve. In fact, most EGSnrc transport settings probably don't matter at that point!

For the parameters you suggest (what is EPI?), I would suggest to look at electron microscopy (EM); transmission EM if you can reduce the silicon thickness.

**Reply by Jaap1957** [reply]
The distribution of EGSnrc contains density files up to 1E4 MeV
HEN_HOUSE\pegs4\density_corrections\elements\silicon.density 
has as last enrties:
 6.00E+03,1.431E+01  7.00E+03,1.462E+01  8.00E+03,1.488E+01  9.00E+03,1.512E+01
 1.00E+04,1.533E+01

so it ends with 1e4 = 10GeV.

**Comment by Jaap1957** [comment]
Another question:
Now I want to work at 2GeV. 

When i try to set UP and or UE  > 999 MeV, they fall to the defaults without error mention.
In the input file I set

:start media definition: # Only 1 media definition block allowed

    UE = 1000.0 # maximum energy for electrons (kinetic+0.511)
    UP = 1000.0 # maximum energy for photons (kinetic)
    AE = 0.512 # lowest energy for electrons (kinetic+0.511)
    AP = 0.001 # lowest energy for photons (kinetic)

which results in output:
....
  PEGSLESS INPUT.  CALCULATING ELECTRON CROSS-SECTIONS.
  density_file silicon                                                                                                                                                                                

  Medium data:

 AE =    5.12000E-01 MeV,  UE =    5.05110E+01 MeV
 AP =    1.00000E-03 MeV,  UP =    5.00000E+01 MeV

When I set
    UE = 999.0 # maximum energy for electrons (kinetic+0.511)
    UP = 999.0 # maximum energy for photons (kinetic)

output reads
  Medium data:

 AE =    5.12000E-01 MeV,  UE =    9.99000E+02 MeV
 AP =    1.00000E-03 MeV,  UP =    9.99000E+02 MeV

so it happens above 999

Where can I override the 999 limit?

**Reply by ftessier** [reply]
That is because the density effect correction files distributed with EGSnrc are generated with energies up to 1 GeV. Above that you will need to [generate the density effect corrections](https://github.com/nrc-cnrc/EGSnrc/wiki/Generating-density-effect-correction-files-with-ESTAR) for the materials you need, adding energies above 1 GeV.

**Reply by Jaap1957** [reply]
Thanks again !

I followed this, and found the distribution already contains a file up to 1e4 MeV.
HEN_HOUSE\pegs4\density_corrections\elements\silicon.density 

The problem persists.

**Reply by ftessier** [reply]
Oops, you are right, I misread the energy value in the density correction file! Then it must be one of the other cross section files in `$HEN_HOUSE/data`. @mainegra, do you recall which file limits energy to 1 GeV out of the box? I already got it wrong once 😄

**Reply by mainegra** [reply]
I looked into this a month ago to [answer a user question on reddit](https://www.reddit.com/r/EGSnrc/comments/rho6p7/gev_range_in_egsnrc/). It seems NIST ESTAR is limited to 10 GeV. The same limit is imposed by the _pegsless_ input routines in EGSnrc, probably because of its dependency with density correction files. I would suggest we investigate this issue (make it a github issue) further! Perhaps a good project for a co-op student? In the Medical Physics field, transporting photons and electrons at higher energies has not been the focus in the last 20 years.

**Reply by ftessier** [reply]
These maximum values are set in the `get_media_inputs.mortran` source file starting on [line 814](https://github.com/nrc-cnrc/EGSnrc/blob/a6fc389c6465c949645380976f38a013a639759c/HEN_HOUSE/src/get_media_inputs.mortran#L814-L848). @blakewalters do you know why these maximum values are set so "low", and if it is safe to increase them?

**Reply by mainegra** [reply]
BTW, All the other cross section data files in EGSnrc seem to go all the way to 100 GeV or more ...

**Reply by ftessier** [reply]
Ah ok thanks for the answer @mainegra!

**Reply by mainegra** [reply]
@ftessier as I said, the limit in _pegsless_ is because it relies on the use of density correction files to define media!!!

**Reply by ftessier** [reply]
Yes, understood, we were typing at the same time, you answered just before I asked; very efficient support! 😄

**Reply by Jaap1957** [reply]
> These maximum values are set in the `get_media_inputs.mortran` source file starting on [line 814](https://github.com/nrc-cnrc/EGSnrc/blob/a6fc389c6465c949645380976f38a013a639759c/HEN_HOUSE/src/get_media_inputs.mortran#L814-L848). @blakewalters do you know why these maximum values are set so "low", and if it is safe to increase them?

Again many thanks Mr Tessier !
I have changed the 999 entries  in
\HEN_HOUSE\src\get_media_inputs.mortran
at line 838 and 847 to 20200 (20GeV).
now the simulation runs fine, and i can start re calculating these old data ...

**Comment by Jaap1957** [comment, marked as answer]
To summarize the remakrs of Mr Townson and Mr Tessier:

To add radiative compton corrections, follow the instructions here:
https://www.reddit.com/r/EGSnrc/comments/ecuql7/radiative_compton_corrections_on/
The result is a Makefile like the attached file (remove the .txt)  (this one is in egs_phd):
[Makefile.txt](https://github.com/nrc-cnrc/EGSnrc/files/7941569/Makefile.txt)

For simulations above 1 GeV, change the 999 entries in`\HEN_HOUSE\src\get_media_inputs.mortran` at [lines 838 and 847](https://github.com/nrc-cnrc/EGSnrc/blob/master/HEN_HOUSE/src/get_media_inputs.mortran#L838,L847) to, for example, 10100 (10 GeV).

The pegs4 tables at `HEN_HOUSE\pegs4\density_corrections\...` reach up to 10GeV.

**Reply by ftessier** [reply]
Note that documentation about turning on radiative Compton corrections are also found in the [EGSnrc manual](https://nrc-cnrc.github.io/EGSnrc/doc/pirs701-egsnrc.pdf), Section `2.2.2.iii` (page 45).

**Comment by ftessier** [comment]
Further discussion in #831.
